data-structures/queue/queue.py

```diff
@@ -0,0 +1,45 @@
+from node import Node
+
+class Queue:
+    def __init__(self, iter=()):
+        self.head = None
+        self._size = 0
+        for el in iter:
+            self.enqueue(el)
+
+    def __str__(self):
+        """Returns string with Queue informations(head, length)"""
+        if self.head:
+            return 'Stacks First value is {} and length is {}\
+            '.format(self.head.val, self._size)
+        return 'Stack Empty'
+
+    def __len__(self):
+        """Returns size of Queue Instance"""
+        return self._size
+
+    def enqueue(self, val):
+        """Appends to the end of the line"""
+        self.head = Node(val, self.head)
+        self._size += 1
+
+    def dequeue(self):
+        """Pops front node of Queue and returns it"""
+        if not self.head:
+            raise IndexError('')
```

Comment:
When you raise an exception like this make sure you give an explanation in this string instead of leaving it empty.